From 50d6921f9089174042a5f3af6d3cc7f6d24ebfa2 Mon Sep 17 00:00:00 2001
From: akodanka <anoob.anto.kodankandath@intel.com>
Date: Wed, 24 Jun 2020 20:28:19 +0530
Subject: [PATCH] USB Audio should have a higher priority over HDMI

HDMI gets detected as AUDIO_DEVICE_OUT_LINE device.
Since this LINEOUT has a higher priority than USB, this gets
picked above USB Headset from the audio policy for both media
and call scenarios.

Hence, changing the priority of these devices to pick USB audio
over LINEOUT to correct the behavior.

Tracked-On: OAM-91514
Signed-off-by: akodanka <anoob.anto.kodankandath@intel.com>
---
 services/audiopolicy/enginedefault/src/Engine.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/services/audiopolicy/enginedefault/src/Engine.cpp b/services/audiopolicy/enginedefault/src/Engine.cpp
index 04170ac3ee..729898c750 100644
--- a/services/audiopolicy/enginedefault/src/Engine.cpp
+++ b/services/audiopolicy/enginedefault/src/Engine.cpp
@@ -248,8 +248,6 @@ audio_devices_t Engine::getDeviceForStrategyInt(legacy_strategy strategy,
             if (device) break;
             device = availableOutputDevicesType & AUDIO_DEVICE_OUT_WIRED_HEADSET;
             if (device) break;
-            device = availableOutputDevicesType & AUDIO_DEVICE_OUT_LINE;
-            if (device) break;
             device = availableOutputDevicesType & AUDIO_DEVICE_OUT_USB_HEADSET;
             if (device) break;
             device = availableOutputDevicesType & AUDIO_DEVICE_OUT_USB_DEVICE;
@@ -264,6 +262,8 @@ audio_devices_t Engine::getDeviceForStrategyInt(legacy_strategy strategy,
                 device = availableOutputDevicesType & AUDIO_DEVICE_OUT_ANLG_DOCK_HEADSET;
                 if (device) break;
             }
+            device = availableOutputDevicesType & AUDIO_DEVICE_OUT_LINE;
+            if (device) break;
             device = availableOutputDevicesType & AUDIO_DEVICE_OUT_EARPIECE;
             break;
 
@@ -425,9 +425,6 @@ audio_devices_t Engine::getDeviceForStrategyInt(legacy_strategy strategy,
         if (device2 == AUDIO_DEVICE_NONE) {
             device2 = availableOutputDevicesType & AUDIO_DEVICE_OUT_WIRED_HEADPHONE;
         }
-        if (device2 == AUDIO_DEVICE_NONE) {
-            device2 = availableOutputDevicesType & AUDIO_DEVICE_OUT_LINE;
-        }
         if (device2 == AUDIO_DEVICE_NONE) {
             device2 = availableOutputDevicesType & AUDIO_DEVICE_OUT_WIRED_HEADSET;
         }
@@ -443,6 +440,9 @@ audio_devices_t Engine::getDeviceForStrategyInt(legacy_strategy strategy,
         if (device2 == AUDIO_DEVICE_NONE) {
             device2 = availableOutputDevicesType & AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET;
         }
+        if (device2 == AUDIO_DEVICE_NONE) {
+            device2 = availableOutputDevicesType & AUDIO_DEVICE_OUT_LINE;
+        }
         if ((device2 == AUDIO_DEVICE_NONE) && (strategy != STRATEGY_SONIFICATION)) {
             // no sonification on aux digital (e.g. HDMI)
             device2 = availableOutputDevicesType & AUDIO_DEVICE_OUT_AUX_DIGITAL;
-- 
2.17.1

